maths/manhattan_distance.py

```diff
@@ -0,0 +1,133 @@
+from typing import Any
+
+
+def manhattan_distance(point_a: list, point_b: list) -> float:
+    """
+    Expectts two list of numbers representing two points in the same
+    n-dimensional space
+
+    https://en.wikipedia.org/wiki/Taxicab_geometry
+
+    >>> manhattan_distance([1,1], [2,2])
+    2.0
+    >>> manhattan_distance([1.5,1.5], [2,2])
+    1.0
+    >>> manhattan_distance([1.5,1.5], [2.5,2])
+    1.5
+    >>> manhattan_distance([-3, -3, -3], [0, 0, 0])
+    9.0
+    >>> manhattan_distance([1,1], None)
+    Traceback (most recent call last):
+         ...
+    ValueError: Missing an input
+    >>> manhattan_distance([1,1], [2, 2, 2])
+    Traceback (most recent call last):
+         ...
+    ValueError: Both points must be in the same n-dimensional space
+    >>> manhattan_distance([1,"one"], [2, 2, 2])
+    Traceback (most recent call last):
+         ...
+    TypeError: Expected a list of numbers as input, found str
+    >>> manhattan_distance(1, [2, 2, 2])
+    Traceback (most recent call last):
+         ...
+    TypeError: Expected a list of numbers as input, found int
+    >>> manhattan_distance([1,1], "not_a_list")
+    Traceback (most recent call last):
+         ...
+    TypeError: Expected a list of numbers as input, found str
+    """
+
+    _validate_point(point_a)
+    _validate_point(point_b)
+    if len(point_a) != len(point_b):
+        raise ValueError("Both points must be in the same n-dimensional space")
+
+    total_distance = 0.0
+    for i in range(len(point_a)):
+        total_distance += abs(point_a[i] - point_b[i])
+
+    return total_distance
+
+
+def _validate_point(point: Any) -> None:
```

Comment:
```suggestion
def _validate_point(point: list[float]) -> None:
```

Reply:
Cause being a validation function I expect anything as input and inside I do the checks.

Reply:
That is not the way that type hints work.